### PR TITLE
Back out of compiling queries if javac is unavailable

### PR DESCRIPTION
**Pull request by thobe**

### CODE REVIEW

File: community/codegen/src/main/java/org/neo4j/codegen/CodeGenerationStrategy.java [hunk 1]
```diff
@@ -76,8 +87,10 @@ public void addByteCodeVisitor( ByteCodeVisitor visitor )
         }
 
         CodeGenerator generateCode( ClassLoader loader, CodeGeneratorOption[] options )
+                throws CodeGenerationNotSupportedException
         {
-            CodeGenerator generator = strategy.generateCode( loader, options );
+            CodeGenerator generator;
+            generator = strategy.generateCode( loader, options );
```
Comment (lutovich) [inline community/codegen/src/main/java/org/neo4j/codegen/CodeGenerationStrategy.java]: I think declaration and assignment should be on the same line

Comment (thobe) [inline community/codegen/src/main/java/org/neo4j/codegen/CodeGenerationStrategy.java]: So do I... How did that come to look this way?

Right... I tried handling the exception here, by wrapping and re-throwing, but then I just changed the exception hierarchy instead...

File: community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/codegen/CodeStructure.scala [hunk 2]
```diff
@@ -155,7 +156,11 @@ trait MethodStructure[E] {
 object CodeStructure {
   val __TODO__MOVE_IMPLEMENTATION = new CodeStructure[GeneratedQueryExecution] {
     override def generateQuery(packageName: String, className: String, columns: Seq[String], opIds: Seq[String])(block: MethodStructure[_] => Unit) = {
-      val generator = codegen.CodeGenerator.generateCode(CodeStructure.getClass.getClassLoader/*, SourceCode.PRINT_SOURCE*/)
+      val generator = try {
+        codegen.CodeGenerator.generateCode(CodeStructure.getClass.getClassLoader/*, SourceCode.PRINT_SOURCE*/)
+      } catch {
+        case e: Exception => throw new CantCompileQueryException(e.getMessage, e)
```
Comment (lutovich) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/codegen/CodeStructure.scala:162]: Does this mean that we'll fallback to interpreted runtime on any error and not only on `CodeGenerationNotSupportedException`?

Comment (thobe) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/codegen/CodeStructure.scala:162]: yes, any Exception thrown while trying to create a `CodeGenerator`

**Pull request merged**